packages/xo-server-perf-alert/src/Rules.js

```diff
@@ -0,0 +1,183 @@
+// @ts-check
+
+/**
+ * @import {XoHost, XoSr, XoVm} from "@vates/types"
+ */
+
+/**
+ *
+ * @param {XoSr} sr
+ * @returns {boolean}
+ */
+function isSrWritable(sr) {
+  return sr !== undefined && sr.content_type !== 'iso' && sr.size > 0
+}
+
+export class MonitorRule {
+  /**
+   * @type {'>' | '<'}
+   */
+  comparator
+
+  /**
+   * @type {'host' | 'SR' | 'VM'}
+   */
+  objectType
+  /**
+   * @type {number}
+   */
+  triggerLevel
+
+  /**
+   * @type {number}
+   */
+  triggerPeriod
+  /**
+   * @type {boolean}
+   */
+  smartMode
+
+  /**
+   * @type {boolean}
+   */
+  excludeUuids
+
+  /**
+   * @type {Array<string>}
+   */
+  uuids
+
+  /**
+   * @type {'cpuUsage' | 'memoryUsage' | 'storageUsage'}
+   */
+  variableName
+
+  /**
+   * @returns {string}
+   */
+  get id() {
+    return `${this.objectType.toLocaleLowerCase()}|${this.variableName}|${this.triggerLevel}`
+  }
+  /** */
+  constructor({
+    alarmTriggerLevel,
+    alarmTriggerPeriod = 60,
+    excludeUuids,
+    comparator,
+    objectType,
+    smartMode,
+    uuids,
+    variableName,
+  }) {
+    this.triggerLevel = alarmTriggerLevel
+    this.triggerPeriod = alarmTriggerPeriod
+    this.excludeUuids = excludeUuids
+    this.comparator = comparator
+    this.objectType = objectType
+    this.smartMode = smartMode
+    this.uuids = uuids
+    this.variableName = variableName
+  }
+
+  /**
+   *
+   * @param {XoVm|XoHost|XoSr} xoObject
+   * @returns {boolean}
+   */
+  isObjectAffected(xoObject) {
+    if (this.smartMode || this.excludeUuids) {
+      if (xoObject.type !== this.objectType) {
+        return false
+      }
+      if (xoObject.type === 'SR' && !isSrWritable(xoObject)) {
+        return false
+      }
+      if (xoObject.type === 'VM' || xoObject.type === 'host') {
+        if (xoObject.power_state !== 'Running') {
+          return false
+        }
+      }
+      if (this.excludeUuids && this.uuids.includes(xoObject.uuid)) {
+        return false
+      }
+    } else {
+      if (!this.uuids.includes(xoObject.uuid)) {
+        return false
+      }
+    }
+
+    return true
+  }
```

Comment:
What do you think?
Isn't easier to read?

```suggestion
  isObjectAffected(xoObject) {
    if (this.smartMode) {
      if (xoObject.type !== this.objectType) {
        return false
      }
      if (xoObject.type === 'SR' && !isSrWritable(xoObject)) {
        return false
      }
      if ((xoObject.type === 'VM' || xoObject.type === 'host') && xoObject.power_state !== 'Running') {
          return false
      }
      
      return true
      }
      
     const includeUuid = this.uuids.includes(xoObject.uuid)
     return this.excludeUuids ? !includeUuid : includeUuid
  }
```